esp_isotp/inc/esp_isotp.h

```diff
@@ -48,6 +87,38 @@
 extern "C" {
 #endif
 
+/**
+ * @brief ISO-TP receive callback function type
+ *
+ * Called when a complete message has been received successfully.
+ *
+ * @warning This callback executes in the same context as isotp_on_can_message(),
+ *          which is typically ISR context. Keep callback execution time minimal
+ *          and avoid blocking operations.
+ * @note The data pointer is only valid during the callback execution.
+ *       Copy data immediately if needed beyond the callback scope.
+ *
+ * @param[in] data Pointer to received data (valid only during callback)
+ * @param[in] size Size of received data in bytes
+ * @param[in] user_arg User argument provided during configuration
+ */
+typedef void (*esp_isotp_rx_callback_t)(const uint8_t *data, uint32_t size, void *user_arg);
```

Comment:
question, are these callbacks also be called in the `esp_isotp_poll`?

Reply:
It's hard for the user to know which isotp link invokes the callback, becuase there's no "ISOTP link handle" in the callback function